### PR TITLE
parts of frontend setup issue #6

### PR DESCRIPTION
**Pull request by wisskirchenj**
contains: 
- suggested folder structure including Readme-description
- simplification and refactoring of the vite-provided plugins and router setup (e.g. splitting routes in separate .ts)
- provide route to CardDetailsPage to be created in #16 (temporary code in Home.vue to be replaced by real DashBoard page later)
- install axios client and oauth-extension
- add runconfig to start the frontend without server

### CODE REVIEW

**Review by codeofcarbon — APPROVED** [review]
played with your changes a bit - LGTM

File: frontend/src/shared/pages/Home.vue [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<template>
+  <HelloWorld />
```
Comment (Taz03) [inline frontend/src/shared/pages/Home.vue]: remove HelloWorld?

File: frontend/src/shared/pages/ErrorNotFound.vue
Comment (Taz03) [inline frontend/src/shared/pages/ErrorNotFound.vue:1]: it is not in fullscreen, and the button have no text/not visible
image

**Review by Taz03 — CHANGES_REQUESTED** [review]

File: frontend/src/shared/pages/ErrorNotFound.vue
Comment (codeofcarbon) [reply frontend/src/shared/pages/ErrorNotFound.vue:1]: I believe that as of now we should treat that page as - 'yes, there is a 404 page' and care less about it current look

Comment (wisskirchenj) [reply frontend/src/shared/pages/ErrorNotFound.vue:1]: yeah it's just a placeholder - but that should not block this PR

File: frontend/src/shared/pages/Home.vue (hunk 1 above)
Comment (wisskirchenj) [reply frontend/src/shared/pages/Home.vue]: yeah - why not?
Replaced it with Splashscreeb

File: frontend/package.json [hunk 2]
```diff
@@ -9,6 +9,8 @@
   },
   "dependencies": {
     "@mdi/font": "7.0.96",
+    "axios": "^1.6.5",
```
Comment (alfabravo2013) [inline frontend/package.json:12]: Not an issue but a question: does frontend need proxying in the dev mode to avoid CORS issues?

**Review by alfabravo2013 — APPROVED** [review]

File: frontend/package.json (hunk 2 above)
Comment (wisskirchenj) [reply frontend/package.json:12]: I am not sure - if I recall correctly we have some minimal CORS configuration in the customer project.
I can check tomorrow when setting up Oauth

**Review by Taz03 — APPROVED** [review]

File: frontend/package.json (hunk 2 above)
Comment (codeofcarbon) [reply frontend/package.json:12]: I am also not fully sure but I would say yes - it is more about browser security features, not related directly to the code and envs. However, I am guessing here

**Pull request merged**